### PR TITLE
Use firefox binary on macOS

### PR DESCRIPTION
**Pull request by willdurand**

### CODE REVIEW

**Review by rpl — APPROVED** [review]
I took a quick look to the diff and this looks a pretty reasonable set of changes to me

**Comment by Rob--W** [comment]
The commit title suggest that firefox-bin is used on macOS only, but this change applies to all platforms. If we generate a changelog, let's try to remove the "macOS" part (or rephrase it to make it clear that it fixes a bug on macOS).

For the record, this patch is needed to resolve https://github.com/mozilla/web-ext/issues/2975 ; details at https://github.com/mozilla/web-ext/issues/2975#issuecomment-1879847209 , which links to https://bugzilla.mozilla.org/show_bug.cgi?id=1871447

**Comment by willdurand** [comment]
ha oops, good call. I fixed the relnotes.

**Comment by fregante** [comment]
Thank you! 🙏
